File: php/features.php
<?php
declare(strict_types=1);

// FreeToPlayGames SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeToPlayGamesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeToPlayGamesBaseFeature();
            case "test":
                return new FreeToPlayGamesTestFeature();
            default:
                return new FreeToPlayGamesBaseFeature();
        }
    }
}

File: py/freetoplaygames_sdk/features.py
# FreeToPlayGames SDK feature factory

from freetoplaygames_sdk.feature.base_feature import FreeToPlayGamesBaseFeature
from freetoplaygames_sdk.feature.test_feature import FreeToPlayGamesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeToPlayGamesBaseFeature(),
        "test": lambda: FreeToPlayGamesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
